Makefile

```diff
@@ -11,13 +11,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Ensure GOBIN is not set during build so that promu is installed to the correct path
+unexport GOBIN
+
 GO           ?= go
 GOFMT        ?= $(GO)fmt
 FIRST_GOPATH := $(firstword $(subst :, ,$(shell $(GO) env GOPATH)))
+GOHOSTOS     ?= $(shell $(GO) env GOHOSTOS)
+GOHOSTARCH   ?= $(shell $(GO) env GOHOSTARCH)
+
+# Enforce Go modules support just in case the directory is inside GOPATH (and for Travis CI).
+export GO111MODULE=on
```

Comment:
Can you use `GO111MODULE := on`? With this change, I don't see travis tries to pull all the packages.
See the PR I created: https://github.com/Stackdriver/stackdriver-prometheus-sidecar/pull/145
And the build with the change I made: https://travis-ci.com/Stackdriver/stackdriver-prometheus-sidecar/builds/122476675

Reply:
Good finding! Somehow that breaks builds under gopath, but Travis takes precedence.